### PR TITLE
Add memory specific changes to LTP

### PR DESCRIPTION
**Pull request by harish-24**
Patch adds necessary changes to run few memory tests which needs setup before actual test execution.

Signed-off-by: Harish <harish@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by harish-24** [comment]
Depends on https://github.com/avocado-framework/avocado/pull/3171

**Comment by harish-24** [comment]
Library patch is merged. Please review.

File: generic/ltp.py [hunk 1]
```diff
@@ -49,45 +89,60 @@ def setUp(self):
...
-        with open(logfile, 'r') as fp:
-            lines = fp.readlines()
+        with open(logfile, 'r') as file_p:
+            lines = file_p.readlines()
             for line in lines:
                 if 'FAIL' in line:
                     value = re.split(r'\s+', line)
                     self.failed_tests.append(value[0])
 
+        collect_dmesg(self)
```
Comment (PraveenPenguin) [inline generic/ltp.py]: i think it is good if we clear dmesg also before running any test

File: generic/ltp.py [hunk 2]
```diff
@@ -19,28 +19,68 @@
...
 from avocado.utils import process, archive
-import os
-import re
+from avocado.utils.partition import Partition
 
 from avocado.utils.software_manager import SoftwareManager
 
 
-class ltp(Test):
+def collect_dmesg(obj):
```
Comment (PraveenPenguin) [inline generic/ltp.py]: and please move this change in other commit

**Comment by harish-24** [comment]
@PraveenPenguin Pushed with changes.

**Pull request merged**